### ISSUE TITLE
Suggestions on Dandelifon's capacity

### ISSUE THREAD

**Issue by kangzaig** [Vazkii/Botania#4292]
### Mod Loader

Forge

### Minecraft Version

1.12.2-1.19

### Botania version

1.12.2-1.19

### Modloader version

forge

### The latest.log file

suggest

### Issue description

Dandelifon's production capacity is too tight, and linear production capacity cannot play the role that life game should have. Dandelifon is reduced to a useless toy. It is suggested to modify the cell block to make it expensive, but try to ensure that it can still be mass produced, and modify to improve the mana output of multiple cell settlement.

**Comment by kangzaig** [comment]
In actual use, one Dandelineon can be used to produce cell blocks, and the other Dandelineon can accept cell blocks, which makes the linear output of life particularly ridiculous. It is also recommended to modify it properly, so that the long-lived cells can obtain higher income.

**Comment by kangzaig** [comment]
Every flower should have its place and uniqueness.

**Comment by cplir-c** [comment]
> one Dandelineon can be used to produce cell blocks, and the other Dandelineon can accept cell blocks

The documentation says that cells in range of multiple dandelifeons die immediately
I suppose you could break and replace dandelifeons so you only have one active at a time

**Comment by kangzaig** [comment]
> > one Dandelineon can be used to produce cell blocks, and the other Dandelineon can accept cell blocks
> 
> The documentation says that cells in range of multiple dandelifeons die immediately I suppose you could break and replace dandelifeons so you only have one active at a time

This may be a feasible way,[](https://www.bilibili.com/video/BV1yP411j7aV).This may be a feasible approach. In fact, it is not important. The important thing is that Dandelineon, as an expensive and interesting flower, has a very low ability to produce mana. The amount of production of mana increases linearly with time, which hides the necessity of life game. I hope to increase the reward of longevity and the price of cell block, so that the challenge of Dandelineon can have a higher use significance.

**Comment by kangzaig** [comment]
> > one Dandelineon can be used to produce cell blocks, and the other Dandelineon can accept cell blocks
> 
> The documentation says that cells in range of multiple dandelifeons die immediately I suppose you could break and replace dandelifeons so you only have one active at a time

This may be a feasible way,[https://www.bilibili.com/video/BV1yP411j7aV].This may be a feasible approach. In fact, it is not important. The important thing is that Dandelineon, as an expensive and interesting flower, has a very low ability to produce mana. The amount of production of mana increases linearly with time, which hides the necessity of life game. I hope to increase the reward of longevity and the price of cell block, so that the challenge of Dandelineon can have a higher use significance.

**Comment by kangzaig** [comment]
> > one Dandelineon can be used to produce cell blocks, and the other Dandelineon can accept cell blocks
> 
> The documentation says that cells in range of multiple dandelifeons die immediately I suppose you could break and replace dandelifeons so you only have one active at a time

This may be a feasible way,[https://www.bilibili.com/video/BV1yP411j7aV].This may be a feasible approach. In fact, it is not important. The important thing is that Dandelineon, as an expensive and interesting flower, has a very low ability to produce mana. The amount of production of mana increases linearly with time, which hides the necessity of life game. I hope to increase the reward of longevity and the price of cell block, so that the challenge of Dandelineon can have a higher use significance.

**Comment by Alwinfy** [comment]
Hi, thanks for taking an interest!

Good news: it turns out that someone else has had similar complaints recently, and [a patch has been submitted](https://lists.sr.ht/~williewillus/violet-moon/patches/38608) to change the behavior! 

The details of the patch are as follows:

1. Cells in *overlapping* Dandelifeon regions still die, BUT
2. Cells in Dandelifeons who are spaced _exactly_ 25 blocks apart can affect each other.
3. Multiple Dandlifeons run at the same tick speed, so you can treat neighboring regions as one big region.
4. When a Dandelifeon eats a cell block, it only clears the blocks in its own region. Other regions are unaffected.
5. However, when a cell block crosses a region boundary (ex. as a Glider), it loses ¾ of its mana value!

This way, if you are smart about placing Dandelifeons, you can get mana for "free." However, extracting "maximum" (100 cycles worth) of mana per iteration is still quite difficult!

The patch is still being reviewed, but it will hopefully be merged soon.

I hope this information is to your satisfaction! Have fun!

**Comment by kangzaig** [comment]
> Hi, thanks for taking an interest!
> 
> Good news: it turns out that someone else has had similar complaints recently, and [a patch has been submitted](https://lists.sr.ht/~williewillus/violet-moon/patches/38608) to change the behavior!
> 
> The details of the patch are as follows:
> 
> 1. Cells in _overlapping_ Dandelifeon regions still die, BUT
> 2. Cells in Dandelifeons who are spaced _exactly_ 25 blocks apart can affect each other.
> 3. Multiple Dandlifeons run at the same tick speed, so you can treat neighboring regions as one big region.
> 4. When a Dandelifeon eats a cell block, it only clears the blocks in its own region. Other regions are unaffected.
> 5. However, when a cell block crosses a region boundary (ex. as a Glider), it loses ¾ of its mana value!
> 
> This way, if you are smart about placing Dandelifeons, you can get mana for "free." However, extracting "maximum" (100 cycles worth) of mana per iteration is still quite difficult!
> 
> The patch is still being reviewed, but it will hopefully be merged soon.
> 
> I hope this information is to your satisfaction! Have fun!



> Hi, thanks for taking an interest!
> 
> Good news: it turns out that someone else has had similar complaints recently, and [a patch has been submitted](https://lists.sr.ht/~williewillus/violet-moon/patches/38608) to change the behavior!
> 
> The details of the patch are as follows:
> 
> 1. Cells in _overlapping_ Dandelifeon regions still die, BUT
> 2. Cells in Dandelifeons who are spaced _exactly_ 25 blocks apart can affect each other.
> 3. Multiple Dandlifeons run at the same tick speed, so you can treat neighboring regions as one big region.
> 4. When a Dandelifeon eats a cell block, it only clears the blocks in its own region. Other regions are unaffected.
> 5. However, when a cell block crosses a region boundary (ex. as a Glider), it loses ¾ of its mana value!
> 
> This way, if you are smart about placing Dandelifeons, you can get mana for "free." However, extracting "maximum" (100 cycles worth) of mana per iteration is still quite difficult!
> 
> The patch is still being reviewed, but it will hopefully be merged soon.
> 
> I hope this information is to your satisfaction! Have fun!

Yes, I think this is very good, and it will be very difficult to produce long-lived cells based on this method. Have you considered appropriately increasing the magic production volume produced by the settlement of long-lived cells?For example, using the appropriate algorithm, cells with 100 lifespan can produce thousands of times as long as cells with 1 lifespan, and the cost of cell squares is appropriately increased.

**Comment by tadhgmister** [comment]
I strongly disagree. Personally I find the Dandelifeon fits perfectly as the end-game generating flower **for casual players.**

The linear mana production with age means that any viable sequence that gets the desired 6 cell consumption will perform equally in terms of mana per second assuming the reload time can be gotten negligibly low. Getting a longer sequence means you have to reload it less often which can improve resource efficiency and does improve the production when reload time is not instant but that isn't a huge deal. The low cost of cell blocks makes it easy to play with the setup manually until a desirable one is found and then takes extremely low resources to run continuously **with very few ways for the setup to break itself.**

It takes about 14 seconds of Dandelifeon simulation for 6 cell blocks to fill a diluted mana pool. Compare this with a few other generating flora:
- A Kekimurus takes ~22 seconds to fill a diluted pool (60% longer) and takes a lot more infrastructure to automate but is similarly safe if time consuming to setup, easy to scale up, and easy to prevent jamming once running.
- A Munchdew takes 12.5 seconds to fill a diluted pool (bit faster if never run out of leaves) and is similarly automated with resources that you are likely wanting to automate anyway (wood / crops) but it is much harder to get it to run continuously and can jam if you run out of saplings
- A Entropinnyum can run as fast as you feel safe doing and similarly needs careful consideration to make the initial setup (and is often better to just settle for a sub-optimal but clearly workable one,) but it needs much harder to automate ingredients (sand and gunpowder) and are obviously much more dangerous

I'm not sure what flowers you are comparing the Dandelifeon to but it seems like a very reasonable upgrade to me. If the cost of the cellular blocks were increased then it would just mean players would be less likely to use them just playing with the game of life for the fun of it and finding a suitable solution. With a higher resource cost I feel like players would feel much more pressured to find setups that either use other modded tools to pick up the cell blocks or use multiple flowers to simulate between them to bypass the cost. With more-than-linear mana feedback with age it means players would feel more pressure to get exactly 100 or extremely close to it instead of just settling for a length that will still consume less of their excess ingredients they are happy to spare.

If you want a flower to churn out mana like theres no tomorrow just use a shulk-me-not. If you can find a setup that reliably never runs out of shulkers it is only limited by how fast you can pull the mana out of it. But when a casual player gets to the point where they have gaia shards they are likely near the end of their botania experience and are more likely interested in a low stress puzzle as a cooldown than setting up the next upgrade for mana generation. To me it feels like the perfect epilogue of a casual player's experience, I know it certainly was for me.
